plugin/ochttp/stats_transport.go

```diff
@@ -103,7 +135,22 @@ func (t *tracker) end() {
 	})
 }
 
+func (t *tracker) endServe() {
+	t.endOnce.Do(func() {
+		m := []stats.Measurement{
+			ServerLatency.M(float64(time.Since(t.start)) / float64(time.Millisecond)),
+			ServerResponseBytes.M(t.respSize),
+		}
+		if t.reqSize >= 0 {
+			m = append(m, ServerRequestBytes.M(t.reqSize))
```

Comment:
Minor nit but the asymmetry between when we record the response size (always, even if there is not response body as in a 204 response) and when we record the request body (only if size>0) seems a little odd.

Reply:
I don't disagree, but this was cargo-culted from the roundtripper above, so it at least has the virtue of consistency ;-)

I don't see a meaningful difference between not recording a size at all and recording a size of zero.